src/status_im/infra/events.cljs

```diff
@@ -0,0 +1,25 @@
+(ns status-im.infra.events
+  (:require
+    [utils.re-frame :as rf]))
+
+;; Log error message and call error handle event if any
+(rf/reg-event-fx :infra.rpc/log-error
+ (fn [_ [{:keys [method params error-handler-event]} error]]
+   (merge
+    {:effects.log/warn [(str "Failed to  call rpc: " method)
+                        {:method method
+                         :params params
+                         :error  error}]}
+    (when error-handler-event
+      {:fx [[:dispatch [error-handler-event error]]]}))))
+
+(rf/reg-event-fx :infra.rpc/call
+ (fn [_ [{:keys [method params error-handler-event]}]]
+   ;; :json-rpc/call should be moved to infra layer
```

Comment:
This comment I think shouldn't be here because it's like a TODO. Better to create an issue and label it as `tech-debt` for instance, which is our guideline agreement.

Reply:
I'm actually not sure what the best approach is here.
If we're talking about a significant number of changes, creating an issue for every small one might get a bit tedious.
My idea was to mark places that are clearly in need of change with TODOs, so that whoever encounters them during their work can address them along the way.
But I also see the possible downside - stale TODOs. Still they are easier to create and to remove.
Wdyt?